Core/clim-basic/extended-streams/stream-input.lisp

```diff
@@ -179,58 +179,72 @@
          (when ,old-stream
            (stream-set-input-focus ,old-stream))))))
 
+
+;;; 22.1 Basic Input Streams
 
-;;; Streams are subclasses of standard-sheet-input-mixin regardless of whether
-;;; or not we are multiprocessing.  In single-process mode the blocking calls to
-;;; stream-read-char, stream-read-gesture are what cause process-next-event to
-;;; be called.  It's most convenient to let process-next-event queue up events
-;;; for the stream and then see what we've got after it returns.
+;;; Basic input streams are character streams. It should not happen
+;;; that input-buffer contains anything else than characters. This
+;;; class is disjoint to the extended-input-stream protocol.
+;;;
+;;; Part of the extended input stream protocol is mixed in thanks to
+;;; the input-stream-kernel (STREAM-INPUT-BUFFER, STREAM-SCAN-POINTER
+;;; and STREAM-INSERTION-POINTER accessors and :INPUT-BUFFER
```

Comment:
I've removed the part in parenthesis. It was mentioned, because I've initially implemented something akin input-editing for the input kernel, but then I've decided that it would only harm clarity and duplicate functionality of input-editing-streams.